### PR TITLE
MM-18464 Interactive dialog renders out of safe area view on landscape orientation

### PR DESCRIPTION
**Pull request by imisshtml**
#### Summary
Updated Interactive Dialog for support of SafeAreaView on iOS

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-18464

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes - SafeAreaView padding in landscape on iOS

#### Device Information
This PR was tested on:
iPhone X 12.4 (Simulator)

#### Screenshots
Simulator Screen Shot - iPhone X - 2019-09-26 at 23 12 38
Simulator Screen Shot - iPhone X - 2019-09-26 at 23 12 44
Simulator Screen Shot - iPhone X - 2019-09-26 at 23 12 31

### CODE REVIEW

File: app/components/widgets/settings/bool_setting.js [hunk 1]
```diff
@@ -104,21 +104,23 @@ export default class BoolSetting extends PureComponent {
         }
 
         return (
-            <View style={padding(isLandscape)}>
-                {labelContent}
+            <View>
```
Comment (enahum) [inline app/components/widgets/settings/bool_setting.js]: This View is extra you can change it to a <React.Fragment>

File: app/screens/selector_screen/selector_screen.js [hunk 2]
```diff
@@ -322,7 +325,7 @@ export default class SelectorScreen extends PureComponent {
         return (
             <View style={style.container}>
                 <StatusBar/>
-                <View style={style.searchBar}>
+                <View style={[style.searchBar, padding(true)]}>
```
Comment (enahum) [inline app/screens/selector_screen/selector_screen.js]: why true here?

File: app/screens/selector_screen/selector_screen.js [hunk 3]
```diff
@@ -353,6 +356,7 @@ export default class SelectorScreen extends PureComponent {
                     onRowPress={this.handleSelectItem}
                     renderItem={rowComponent}
                     theme={theme}
+                    isLandscape={true}
```
Comment (enahum) [inline app/screens/selector_screen/selector_screen.js]: same here

**Comment by enahum** [comment]
@imisshtml you have eslint errors

```
> eslint --ignore-path .gitignore --ignore-pattern node_modules --quiet .


/home/circleci/mattermost-mobile/app/components/custom_list/option_list_row/option_list_row.js
  44:24  error  Missing trailing comma  comma-dangle

/home/circleci/mattermost-mobile/app/screens/selector_screen/selector_screen.js
  37:8   error  There should be no space after '{'       object-curly-spacing
  37:10  error  'isLandscape' is defined but never used  no-unused-vars
  37:22  error  There should be no space before '}'      object-curly-spacing
```

File: app/components/autocomplete_selector/autocomplete_selector.js [hunk 4]
```diff
@@ -15,6 +15,7 @@ import {preventDoubleTap} from 'app/utils/tap';
 import {makeStyleSheetFromTheme, changeOpacity} from 'app/utils/theme';
 import {ViewTypes} from 'app/constants';
 import {goToScreen} from 'app/actions/navigation';
+import {paddingHorizontal as padding} from 'app/components/safe_area_view/iphone_x_spacing';
```
[the remark this answers is not in the document]
Comment (imisshtml) [reply app/components/autocomplete_selector/autocomplete_selector.js]: Makes sense, thanks @enahum !

File: app/components/widgets/settings/bool_setting.js (hunk 1 above)
Comment (imisshtml) [reply app/components/widgets/settings/bool_setting.js]: Good catch, thank you!

File: app/screens/selector_screen/selector_screen.js (hunk 2 above)
Comment (imisshtml) [reply app/screens/selector_screen/selector_screen.js]: This was my mistake, thanks for catching it @enahum

**Comment by lindalumitchell** [comment]
Okay with QA to test this one after merge for the sake of expediency (Miguel let me know the build-apps label isn't working at the moment).

**Pull request merged**